src/components/account/account.test.ts

```diff
@@ -0,0 +1,182 @@
+import jwt from 'jsonwebtoken';
+import nock, {RequestBodyMatcher} from 'nock';
+import pino = require('pino');
+import {IUaaUser} from '../../lib/uaa';
+
+import * as uaaData from '../../lib/uaa/uaa.test.data';
+import {IContext} from '../app';
+import {config} from '../app/app.test.config';
+import {Token} from '../auth';
+import * as account from './account';
+
+describe('account test suite', () => {
+  let ctx: IContext;
+
+  describe('account.use-google-sso.view', () => {
+
+    describe('non-GDS users', () => {
+      beforeEach(() => {
+        ctx = setUpUAA(uaaData.user);
+      });
+
+      it('should redirect back to home', async () => {
+        const response = await account.getUseGoogleSSO(ctx, {});
+        expect(response.redirect).toEqual('admin.home');
+        expect(response.body).toBeFalsy();
+      });
+    });
+
+    describe('GDS users', () => {
+      beforeEach(() => {
+        ctx = setUpUAA(uaaData.gdsUser);
+      });
+
+      it('should contain an explanation of the process, and a link to the docs', async () => {
+        const response = await account.getUseGoogleSSO(ctx, {});
+        expect(response.body)
+          .toContain('https://docs.cloud.service.gov.uk/get_started.html#use-the-single-sign-on-function');
+        expect(response.body).toContain('id="sso-process-explanation"');
+      });
+    });
+
+    describe('SSO users', () => {
+      beforeEach(() => {
+        ctx = setUpUAA(uaaData.ssoUser);
+      });
+
+      it('should contain an explanation of the process for opting out', async () => {
+        const response = await account.getUseGoogleSSO(ctx, {});
+        expect(response.body).toContain('id="opt-out-process-explanation"');
+      });
+    });
+  });
+
+  describe('account.use-google-sso.post', () => {
+    beforeEach(() => {
+      ctx = setUpUAA(uaaData.user);
+    });
+
+    it('should set the logged in user\'s origin to google', async () => {
+      const isOriginGoogle: RequestBodyMatcher = (body) => body.origin === 'google';
+      nock(ctx.app.uaaAPI).persist()
+        .put(`/Users/${uaaData.userId}`, isOriginGoogle).reply(200, uaaData.userId);
+
+      await account.postUseGoogleSSO(ctx, {});
+      expect(nock.isDone()).toBeTruthy();
+    });
+
+    it('should render a success page', async () => {
+      nock(ctx.app.uaaAPI).persist()
+        .put(`/Users/${uaaData.userId}`).reply(200, uaaData.userId);
+
+      const response = await account.postUseGoogleSSO(ctx, {});
+      expect(response.body).toContain('Successful');
+    });
+  });
+
+  describe('AccountUser', () => {
+
+    describe('name', () => {
+      it('name combines user\'s given and family names', () => {
+        // We don't have a full implementation of this type
+        // because it's used to represent a JSON payload
+        const uaaUser = {} as any as IUaaUser;
+        uaaUser.name = {
+          givenName: 'User',
+          familyName: 'Name',
+        };
```

Comment:
Don't really like `as any as IUaaUser`. I'd would do:

```
const uaaUser: any = {
  name: {
    givenName: 'User',
    familyName: 'Name',
  }
};
```

... and leave it as the `any` type. It's only a test, and downcasting it back to `IUaaUser` doesn't really buy you any type safety.